### PR TITLE
Set config variable to turn off mixed case for osx-arm64

### PR DESCRIPTION
**Pull request by erykoff**
Checklist
* [x ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

### CODE REVIEW

**Comment by erykoff** [comment]
@conda-forge-admin, please rerender

**Comment by erykoff** [comment]
cc @isuruf

**Comment by isuruf** [comment]
@conda-forge-admin, rerender

**Comment by mingwandroid** [comment]
It might be good to do this for Intel osx too if it doesnt happen by default (when run on a case sensitive fs).

**Comment by erykoff** [comment]
@conda-forge-admin, rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -8,7 +8,7 @@ if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
     export BUILD_CC=${CC_FOR_BUILD}
 fi
 
-if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1"  && "$target_platform" == "osx-arm64" ]]; then
+if [[ "$target_platform" == "osx-64" || "$target_platform" == "osx-arm64"]]; then
```
Comment (mingwandroid) [inline recipe/build.sh]: missing a space here.

Comment (mingwandroid) [reply recipe/build.sh]: in fact:

` if [[ $target_platform == osx-64 ]] || [[ $target_platform == osx-arm64 ]]; then`

**Comment by erykoff** [comment]
@mingwandroid Is this good to go?  I'm hoping to get this in before the release of the new M1 macs next week, otherwise python readline won't work correctly.

**Comment by mingwandroid** [comment]
Yes, thanks!

**Pull request merged**